ideascube/serveradmin/catalog.py

```diff
@@ -296,6 +303,86 @@ def remove(self, install_dir):
         except FileNotFoundError as e:
             printerr(e)
 
+class StaticSite(SimpleZipPackage):
+    typename = 'static-site'
+    handler = Nginx
+
+
+class ZippedMedia(SimpleZipPackage):
+    typename = 'zipped-media'
+    handler = MediaCenter
+
+    def remove(self, install_dir):
+        # Easy part here. Just delete documents from the package.
+        Document.objects.filter(package_id=self.id).delete()
+        super().remove(install_dir)
+
+    def install(self, download_path, install_dir):
+        super().install(download_path, install_dir)
+        print('Adding medias to mediacenter database.')
+        root = self.get_root_dir(install_dir)
+        manifestfile = Path(root, 'manifest.yml')
+        with manifestfile.open('r') as m:
+            manifest = yaml.safe_load(m.read())
+
+        catalog_path = os.path.join(settings.MEDIA_ROOT, "catalog")
+        try:
+            os.symlink(install_dir, catalog_path)
+        except FileExistsError:
+            if not os.path.islink(catalog_path):
+                printerr("Cannot install package {}. {} must not exist "
+                         "or being a symlink.".format(self.id, catalog_path))
+                return
+
+        pseudo_install_dir = os.path.join(catalog_path, self.id)
+        for media in manifest['medias']:
+            try:
+                self._install_media(media, pseudo_install_dir)
+            except:
+                # This can lead to installed package with uninstall media.
+                # We sould handle this somehow.
+                printerr("Cannot install media {} from package {}".format(
+                    media['title'], self.id))
+                continue
+
+    def _install_media(self, media_info, pseudo_install_dir):
+        try:
+            media_info['title'] = title = smart_truncate(media_info['title'])
+        except KeyError:
+            raise InvalidPackageContent('Missing title in {}'.format(
+                media_info))
+
+        if 'lang' not in media_info:
+            media_info['lang'] = settings.LANGUAGE_CODE
+
+        kind = media_info.get('kind')
+        if not kind or not hasattr(Document, kind.upper()):
+            content_type, _ = mimetypes.guess_type(original)
+            media_info['kind'] = guess_kind_from_content_type(content_type) \
+                or Document.OTHER
+
+        media_info['package_id'] = self.id
+        media_info['original'] = os.path.join(pseudo_install_dir,
+                                              media_info['path'])
+
+        if 'preview' in media_info:
+            media_info['preview'] = os.path.join(pseudo_install_dir,
+                                                 media_info['preview'])
+
+        self._save_media(media_info, pseudo_install_dir)
+
+    def _save_media(self, metadata, install_dir):
```

Comment:
I wonder if all this interaction with the database shouldn't be the responsibility of the handler, rather than the package?